core/crud_component.cxx

```diff
@@ -36,8 +36,137 @@
 #include <asio/io_context.hpp>
 #include <asio/steady_timer.hpp>
 
+#include <tl/expected.hpp>
+
 namespace couchbase::core
 {
+std::pair<std::vector<std::byte>, std::error_code>
+serialize_range_scan_create_options(const range_scan_create_options& options)
+{
+    tao::json::value body{};
+    if (options.without_content) {
+        body["key_only"] = true;
+    }
+
+    if (options.collection_id != 0) {
+        body["collection"] = fmt::format("{:x}", options.collection_id);
+    }
+
+    if (std::holds_alternative<range_scan>(options.scan_type)) {
+        const auto& range = std::get<range_scan>(options.scan_type);
+        body["range"] = {
+            { range.start.exclusive ? "excl_start" : "start", base64::encode(range.start.id) },
+            { range.end.exclusive ? "excl_end" : "end", base64::encode(range.end.id) },
+        };
+    } else if (std::holds_alternative<sampling_scan>(options.scan_type)) {
+        const auto& sampling = std::get<sampling_scan>(options.scan_type);
+        body["sampling"] = {
+            { "samples", sampling.limit },
+            { "seed", sampling.seed.value_or(0) },
+        };
+    } else {
+        return { {}, errc::common::invalid_argument };
+    }
+
+    if (options.snapshot_requirements) {
+        const auto& snapshot = options.snapshot_requirements.value();
+        tao::json::value requirements = {
+            { "vb_uuid", std::to_string(snapshot.vbucket_uuid) },
+            { "seqno", snapshot.sequence_number },
+            { "timeout_ms", options.timeout.count() },
```

Comment:
I couldn't see any mention of kvScanTimeout?

Reply:
fixed